whatshap/cli/compare.py

```diff
@@ -995,15 +1022,24 @@ def get_variant_tables(
     return vcfs
 
 
-def get_sample_to_work_on(vcf_readers: List[VcfReader], requested_sample: Optional[str]):
-    all_samples = set()
+def get_sample_names(
+    vcf_readers: List[VcfReader], requested_sample: Optional[str], ignore_name: bool = False
+) -> List[str]:
+    first_samples = []
     sample_intersection = None
     for vcf_reader in vcf_readers:
         if sample_intersection is None:
             sample_intersection = set(vcf_reader.samples)
         else:
             sample_intersection.intersection_update(vcf_reader.samples)
-        all_samples.update(vcf_reader.samples)
+
+        if ignore_name and len(vcf_reader.samples) > 1:
+            raise CommandLineError(
+                "File {!r} contains multiple samples, option --ignore-sample-name not available.".format(
```

Comment:
I’ve stopped using `!r` in cases like these when I want to get quotation marks. It’ll work here at the moment (I think), but `VcfReader.path` can also be a `pathlib.Path`, and then this displays as `File PosixPath('name.vcf') contains multiple samples`.

Reply:
You are absolutely correct. I tested and it did print `File PosixPath('name.vcf') contains multiple samples` for `pathlib.Path` input. Fixed in https://github.com/whatshap/whatshap/pull/335/commits/2ace49ad2cd9794428accac6ec25bb4d4809a79b.